src/Displaz.jl

```diff
@@ -437,6 +437,25 @@ plot3d!(position; kwargs...) = plot3d!(current(), position; kwargs...)
 plot3d(position; kwargs...)  = plot3d(current(), position; kwargs...)
 
 
+#-------------------------------------------------------------------------------
+"""
+    annotation(plotobj::DisplazWindow, coords::Array{T,1}, text::String, label=text::String)
+
+Place a text annotation at given coordinates
+
+If not specified, `plotobj` is the current plot window.  coords is a three element array [x, y, z],
+text is the text to plot and label its Displaz label, note annotation labels don't appear in Displaz's list of data sets
+they are needed however to unload an annotation. If unspecified the label defaults to the annotation string.
+"""
+function annotation(plotobj::DisplazWindow, coords::Array{T,1}, text::String, label=text::String) where {T<:Real}
+    run_displaz(`-script -server $(plotobj.name) -annotation $text $(coords[1]) $(coords[2]) $(coords[3]) -label $label`)
+    nothing
+end
+annotation(coords::Array{T,1}, text::String, label::String) where {T<:Real} = annotation(current(), coords, text, label)
+annotation(coords::Array{T,1}, text::String) where {T<:Real} = annotation(current(), coords, text)
+annotation(text::String, coords::Array{T,1}) where {T<:Real} = annotation(current(), coords, text)
```

Comment:
The `::String` here should be `::AbstractString` (otherwise it won't work for, eg, `SubString` which can be annoying and confusing as that is commonly encountered as output from functions like `split`). Also it might be best to settle on `annotation(coords,text)` and delete `annotation(text,coords)`, I'm not sure it makes sense to have both.

Also a small nitpick: All these instances of `coords::Array{T,1} ... where {T<:Real}`  can be spelled more succinctly as `Vector{<:Real}`.

Reply:
agreed I've changed this